index.js

```diff
@@ -17,50 +17,59 @@ const normalizeEmitter = emitter => {
 	};
 };
 
-module.exports = (emitter, event, options) => {
-	let cancel;
-
-	const ret = new Promise((resolve, reject) => {
-		if (typeof options === 'function') {
-			options = {filter: options};
-		}
+const multiple = (emitter, event, options) => {
+	options = Object.assign({
+		rejectionEvents: ['error'],
+		multiArgs: false,
+		resolveImmediately: false
+	}, options);
 
-		options = Object.assign({
-			rejectionEvents: ['error'],
-			multiArgs: false
-		}, options);
+	if (isNaN(options.count) || options.count < 1) {
+		throw new TypeError('`count` option should be a number greater than 1');
+	}
```

Comment:
This needs to be inside the `new Promise` or return a `Promise.reject()`. Methods that return a promise should always do so, even in the failure case.

Reply:
No. You're wrong. The promise executes a task, does **not** generate it. We need to verify options in the promise generator, we need to verify the options before a task is scheduled.

Reply:
```js
const foo = pThrottle(invaidOptions); // this wouldn't throw
foo (); // this would throw and should NOT
```

Reply:
https://twitter.com/jaffathecake/status/638659818996269056

Reply:
Oh, you're right. Sorry, I just got confused (cuz I work on many things at the same time and my mind is mixed up) :p